src/euphorie/client/browser/webhelpers.py

```diff
@@ -183,15 +183,15 @@ def redirectOnSurveyUpdate(self):
         )
         return True
 
-    @property
-    @memoize
-    def guest_session_id(self):
-        return self.is_guest_account and self.session_id or None
-
     @property
     @memoize
     def session_id(self):
-        return getattr(self.traversed_session.session, 'id', '')
+        traversed_session = self.traversed_session
+        return (
+            getattr(self.traversed_session.session, "id", "")
+            if traversed_session
+            else ""
+        )
```

Comment:
This is over protection:
```
return getattr(self.traversed_session.session, "id", "")
```
seems enough to me.

Reply:
But what if `self.traversed_session` is None? This happens e.g. on the dashboard. If any template that is used outside of a traversed session calls `session_id`, we would get an error.
(I know that such templates should not be calling `session_id` on the first place. So maybe it IS better to fail.

Reply:
I see now that this is different...

Anyway it should be good to:
```
return traversed_session.session.id if traversed_session else ""
```